src/docfx/template/TemplateEngine.cs

```diff
@@ -191,8 +170,61 @@ public JObject PreprocessMetadata(JObject outputMetadata, Document file)
         public JObject TransformTocMetadata(object model)
             => TransformMetadata("toc.json.js", JsonUtility.ToJObject(model));
 
+        private IReadOnlyDictionary<string, (bool isData, bool transformMetadata, (JsonSchemaValidator, JsonSchemaTransformer) jsonSchema)>
+            LoadSchemas(string schemaDir, string contentTemplateDir)
+        {
+            var schemas = new Dictionary<string, (bool isData, bool transformMetadata, (JsonSchemaValidator, JsonSchemaTransformer) jsonSchema)>();
+            var schemaNames = Directory.Exists(schemaDir) ? Directory.EnumerateFiles(schemaDir, "*.schema.json", SearchOption.TopDirectoryOnly)
+                                                    .Select(k => Path.GetFileNameWithoutExtension(k))
+                                                    .Select(k => k.Substring(0, k.Length - ".schema".Length)).ToHashSet() : new HashSet<string>();
```

Comment:
If we only want to remove duplicates here, why not `List.Distict` other than `HashSet`?

Reply:
We only need to enumerate the collection with no further operation on the collection itself, and the order doesn't matter. So I think both HashSet and List are acceptable here.

Reply:
Since we already use Linq here, I think it would be more readable to use `Distinct` in the end instead of doing it implicitly by introducing `HashSet`